californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -731,10 +737,14 @@ private void handleBlock1Response(final Exchange exchange, final Response respon
 
 				clearBlock1Status(key, status);
 
+			} else if (response.getCode() == ResponseCode.REQUEST_ENTITY_TOO_LARGE
+					&& block1.getSize() < status.getCurrentSize()) {
+
+				SendBlockAgain(exchange, response, key, status);
```

Comment:
```
block1.getSize() < status.getCurrentSize()
```

If the blocksize is changed only at the begin, I guess, these checks will have the same result as the above test.
If the blocksize should be changed within the transfer, I'm not sure, when this test should fail and why.

Reply:
I keep it as I consider that this method should be responsible to check this too in case making it more safe to be reused and making it more readable (as the test is explicitly a translation of the RFC sentence)

But the code will maybe change again if we support to turn out a "normal" request (not block1) into a block1 one.

Reply:
My idea was to call sendBlockAgain already above.

Reply:
It could be. I will taking this into account but first I would like to add the _"turn out a "normal" request (not block1) into a block1 one"_ use case.